### ISSUE TITLE
[HOLD] HIGH: [UX Reliability] Incorrect previousReportActionID for a whisper

### ISSUE THREAD

**Issue by roryabraham** [Expensify/App#37908]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Reproducible in staging?:** yes
**Reproducible in production?:** yes
**Issue reported by:** @perunt 
**Slack conversation:** https://expensify.slack.com/archives/C035J5C9FAP/p1709726899587219?thread_ts=1705035404.136629&cid=C035J5C9FAP

## Action Performed:
1. Create a workspace with Account A.
1. Add Account B to the workspace using Account A.
1. Send messages from Account A to Account B in the workspace.
1. Check the messages in the workspace with Account A.
1. Refresh the page

## Expected Result:
All reportActions should form a continuous chain where the previousReportActionID of one points at the reportActionID of the next.

## Actual Result:
One of the whispers breaks the chain.

![image](https://github.com/Expensify/App/assets/47436092/9b041c13-09e8-4e64-b20c-ca5d0192590a)

## Workaround:
@perunt implemented a front-end workaround, but we should still fix this.

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01abf9781cd5d16284</li>
        <li>Upwork Job ID: 1765785466235363328</li>
        <li>Last Price Increase: 2024-03-07</li>
    </ul>
</details>

**Comment by johncschuster** [comment]
@roryabraham, coming from this comment:

> @perunt implemented a front-end workaround, but we should still fix this.

are you implying this will require a back end fix?

**Comment by roryabraham** [comment]
> are you implying this will require a back end fix?

yes, correct

**Comment by johncschuster** [comment]
Sweet. Thanks for clarifying that!

**Comment by johncschuster** [comment]
I chatted with Rory about this, and it looks like this is a part of `Comment Linking` which is a Q1 deliverable. I'm going to raise this to #engineering to get someone to work on it.

**Comment by johncschuster** [comment]
Brought it to [Slack](https://expensify.slack.com/archives/C03SSAQ7P/p1710365608465179) yesterday and have bumped it today

**Comment by johncschuster** [comment]
Bumped it again today

**Comment by johncschuster** [comment]
Bumped again!

**Comment by johncschuster** [comment]
My [thread in #engineering](https://expensify.slack.com/archives/C03SSAQ7P/p1710365608465179) got no traction. I've [raised this to #vip-vsb](https://expensify.slack.com/archives/C066HJM2CAZ/p1712008649292879) and asked for help there.

**Comment by johncschuster** [comment]
Ok! I chatted with @quinthar and confirmed this is a high priority that needs to be delivered in Q1. I've updated the title to reflect the urgency.

**Comment by johncschuster** [comment]
Bumped again!

**Comment by johncschuster** [comment]
I'm going to bump this again Monday so I can get more visibility

**Comment by johncschuster** [comment]
Bumped again in Slack

**Comment by johncschuster** [comment]
@ntdiary Thanks for asking! I'm under the impression this requires an internal engineer to work on it.

**Comment by quinthar** [comment]
Is this still reproducible?

**Comment by quinthar** [comment]
I'm also unclear on the problem.  It says "One of the whispers breaks the chain." -- what does this mean in terms of user impact?

**Comment by johncschuster** [comment]
Great questions! I'm not super familiar with how to validate what the expected behavior or the actual behavior looks like. I've tested a conversation between User A and User B, and the chat seems to be going well.

@perunt, you originally reported this. Are you still able to reproduce the behavior you've reported? If so, can you:

1. clarify why it's a problem, and illuminate what kind of impact it might have.
2. point me to how you validated the errant behavior and in the console, as well as what the correct behavior should look like in the console?

I'm happy to test this further, but I'm not sure I know what a `pass/fail` looks like.

**Comment by roryabraham** [comment]
Pinged [here](https://expensify.slack.com/archives/C035J5C9FAP/p1714430733969299) for clarity and making this a weekly in the meantime.

I think this is probably something we'll want to button up, but I'm not sure off-hand the user impact of the problem - probably just an additional unnecessary network request. Need to dig into this one a bit to uncover more, but need to prioritize May 1st release stuff first. Making this a weekly in the meantime.

**Comment by roryabraham** [comment]
I think I may actually have a proposal that will superfluize this issue: https://expensify.slack.com/archives/C05LX9D6E07/p1714672294012849?thread_ts=1714656403.682529&cid=C05LX9D6E07.

going to throw this on HOLD for a bit while that issue simmers. I confirmed that we have a front-end workaround in place for this for the time being.

**Comment by perunt** [comment]
> I'm also unclear on the problem. It says "One of the whispers breaks the chain." -- what does this mean in terms of user impact?

The issue was that when a whisper message indicating that the user was invited to chat got added to the list, the report actions around it didn't recognize its presence. This created a gap in the user's interaction flow, making it seem like the conversation abruptly ended, even though it hadn't. 

I've seen we've brought back showing whispers for users. I've tested it a few times, and the new whispers have the right `reportActionId` and `previousReportActionId` without causing any issues. But it looks like we'll still need that front-end workaround for older accounts to keep it running smoothly. Just rechecked on the original account and yes, the issue's still there, for obvious reasons.
![Screenshot 2024-05-04 at 11 06 06](https://github.com/Expensify/App/assets/48593211/1d8e698a-6b2b-456b-b4b9-8438dffa733b)
